mlflow/keras/dogs_cats/dag/dag.py

```diff
@@ -0,0 +1,214 @@
+        """
+        Example function that will be performed in a virtual environment.
+
+        Importing at the module level ensures that it will not attempt to import the
+        library before it is installed.
+        """
+        from urllib.request import urlretrieve
+        import os
+        import shutil
+        from airflow.contrib.hooks.gcs_hook import GoogleCloudStorageHook
+
+        run_config = dag_run.conf
+
+        airflow_data_connection = run_config.get('airflow_data_connection', 'wine_input')
+        bucket_name = run_config.get('data_bucket', 'odahu-dev03-data-store')
+
+        tmp_files = set()
+
+        client = GoogleCloudStorageHook(google_cloud_storage_conn_id=airflow_data_connection)
+        if client.exists(bucket_name, dataset_bucket_path):
+            print(f'File {dataset_bucket_path} already exists in bucket {bucket_name}')
+            return
+
+        dataset_url = 'https://download.microsoft.com/download/3/E/1/3E1C3F21-ECDB-4869-8368-6DEBA77B919F/' \
+                      'kagglecatsanddogs_3367a.zip'
+        urlretrieve(dataset_url, filename=dataset_local_filename)
+        tmp_files.add(dataset_local_filename)
+
+        client.upload(bucket=bucket_name, object=dataset_bucket_path, filename=dataset_local_filename)
+
+        for file in tmp_files:
+            print(f'Clean-up: {file}')
+            if os.path.isdir(file):
+                shutil.rmtree(file)
+            os.remove(file)
+
+
+    ensure_dataset_in_bucket = PythonOperator(
+        task_id='ensure_dataset_in_bucket',
+        provide_context=True,
+        python_callable=callable_virtualenv,
+    )
+
+    training_id = 'dogs-cats-classifier'
+    training = ModelTraining(
+        id=training_id,
+        spec=ModelTrainingSpec(
+            model=ModelIdentity(
+                name='dogs-cats-classifier',
+                version='1'
+            ),
+            toolchain='mlflow-project',
+            entrypoint='main',
+            work_dir='mlflow/keras/dogs_cats',
+            hyper_parameters={
+                'epochs': '50'
+            },
+            data=[
+                DataBindingDir(
+                    connection='models-output',
+                    remote_path=dataset_bucket_path,
+                    local_path=f'mlflow/keras/dogs_cats/{dataset_local_filename}'
+                ),
+            ],
+            resources=ResourceRequirements(
+                requests=ResourceList(
+                    cpu='4',
+                    memory='16Gi',
+                    gpu='1'
+                ),
+                limits=ResourceList(
+                    cpu='4',
+                    memory='16Gi',
+                    gpu='1'
+                ),
+            ),
+            algorithm_source=AlgorithmSource(vcs=VCS(
+                connection='odahu-flow-examples',
+                # TODO: switch to develop
+                reference='feat/dogs-cats-example'
```

Comment:
I suppose that it's a good time to switch.
Or are you going to make a separate PR?